pkg/clusters/addons/kong/kong.go

```diff
@@ -146,11 +167,44 @@ func (a *Addon) Deploy(ctx context.Context, cluster clusters.Cluster) error {
 		)
 	}
 
+	if err := clusters.CreateNamespace(ctx, cluster, a.namespace); err != nil {
+		return err
+	}
+
 	// do the deployment and install the chart
 	args := []string{"--kubeconfig", kubeconfig.Name(), "install", DefaultDeploymentName, "kong/kong"}
-	args = append(args, "--create-namespace", "--namespace", a.namespace)
-	args = append(args, a.deployArgs...)
+	args = append(args, "--namespace", a.namespace)
+	if a.enterprise {
+		if a.license != "" && a.license == KongEnterpriseTestLicense {
+			if err := deployKongEnterpriseLicenseSecret(ctx, cluster, a.namespace, KongEnterpriseTestLicense); err != nil {
+				return err
+			}
+		}
+
+		if err := prepareSecrets(ctx, a.namespace); err != nil {
+			return err
+		}
+
+		args = append(args, "--version", "2.3.0", "-f", "https://raw.githubusercontent.com/Kong/charts/main/charts/kong/example-values/minimal-k4k8s-with-kong-enterprise.yaml")
+		license := fmt.Sprintf("enterprise.license_secret=%s", a.license)
+		password := fmt.Sprintf("env.kong_password=%s", a.kongAdminPassword)
+		args = append(args, "--set", "admin.type=LoadBalancer", "--set", "admin.annotations.konghq.com/protocol=http", "--set", "enterprise.rbac.enabled=true",
+			"--set", "env.enforce_rbac=on", "--set", "ingressController.enabled=false",
+			"--set", "ingressController.installCRDs=false", "--set", password,
+			"--skip-crds", "--set", license,
+			// expose new ports
+			"--set", "proxy.stream[0].containerPort=8888",
+			"--set", "proxy.stream[0].servicePort=8888",
+			"--set", "proxy.stream[1].containerPort=9999",
+			"--set", "proxy.stream[1].servicePort=9999",
+			"--set", "proxy.stream[1].parameters[0]=udp",
+			"--set", "proxy.stream[1].parameters[1]=reuseport")
+	} else {
+		args = append(args, a.deployArgs...)
+	}
+
 	stderr = new(bytes.Buffer)
+	fmt.Printf("deployment args %s", args)
```

Comment:
Similar to several previous comments, library code shouldn't print directly to stdout:

```suggestion
```

If you're feeling strongly that this code needs some kind of logging output mechanism, we should add an optional logger implementation.

Reply:
41d7c75fb3875bdbcd50ff3757f5e4ac8c62bb78